### PR TITLE
Add docs for the concept experiment

### LINKED ISSUE #1923: Docs Page "Reference Guide -> Core Concepts -> Experiment"

### PR DESCRIPTION
**Pull request by yeukfu**
close #1923.

### CODE REVIEW

File: docs/user/concept_experiment.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from garage import wrap_experiment
+
+@wrap_experiment
+def my_first_experiment(ctxt=None):
+    print('Hello World!')
+
+my_first_experiment()
+```
+
+Run the example launcher will generate outputs like the following (`CUR_DIR` is
```
Comment (maliesa96) [inline docs/user/concept_experiment.md:22]:
```suggestion
Running the example launcher will generate outputs like the following (`CUR_DIR` is
```

File: docs/user/concept_experiment.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```sh
+2020-08-20 15:18:53 | [my_first_experiment] Logging to CUR_DIR/data/local/experiment/my_first_experiment
+Hello World!
+```
+
+The followings are some useful parameters of `wrap_experiment`. You can see
+details of its parameters [here](../_autoapi/garage/index.html#garage.wrap_experiment).
+
+- log_dir: The custom log directory to log to.
+- snapshot_mode: Policy for which snapshots to keep. The last iteration will be
```
Comment (maliesa96) [inline docs/user/concept_experiment.md:34]: Please specify the acceptable inputs for `snapshot_mode`: `last`, `all`, `none`.

**Comment by maliesa96** [comment]
LGTM aside from the couple of suggestions I made.

**Review by krzentner — APPROVED** [review]
This looks pretty good. We might want to extend it further in the future, but I think we should merge this as is and move on for now.

**Pull request merged**